### ISSUE TITLE
Clean up traits / type parameters.

### ISSUE THREAD

**Issue by c0gent** [poanetwork/hbbft#143]
I would like to add the following supertraits,

```

/// Common traits.
pub mod traits {
	use std::fmt::Debug;
	use std::hash::Hash;

	/// A transaction, user message, etc.
 	pub trait Contribution: Eq + Debug + Hash {}
 	impl<C> Contribution for C where C: Eq + Debug + Hash {}

 	/// A peer node's unique identifier.
 	pub trait NodeUid: Eq + Ord + Clone + Debug + Hash {}
 	impl<N> NodeUid for N where N: Eq + Ord + Clone + Debug + Hash  {}
}
```

and rename current `NodeId` type parameters to `N`, `Tx` to `C`, etc. (as I did in a prior PR somewhere).

**Comment by c0gent** [comment]
Naming of `Contribution` pending agreement, of course.

**Comment by vkomenda** [comment]
Why `Tx` rather than just `T`?

**Comment by c0gent** [comment]
I propose renaming it to `C` (for contribution)... there's no reason to have multiple letter type parameters. Also, as an aside, `Tx` is not a good abbreviation for transaction because it's easily confused with the common (and ubiquitous) abbrev. for transmitter. I always abbreviate transaction as `txn` in code.

**Comment by c0gent** [comment]
Here is the original PR: https://github.com/poanetwork/hbbft/pull/116

**Comment by afck** [comment]
I was thinking of using `C` to distinguish it from `Tx` (or `T` or `Txn`): In `DynamicHoneyBadger` and `HoneyBadger`, the parameter `C` would be the full "contribution" to each epoch, which in practice will often be a collection of transactions. `QueueingHoneyBadger`'s parameter, on the other hand, are the transactions `Tx` themselves, and it will instantiate `DynamicHoneyBadger` with something like `Vec<Tx>` as `C`.

You're right that `Tx` was a bad choice!
I'd be happy with `T`, `C` and `N`, and I like the traits! :+1:

**Comment by c0gent** [comment]
So `NodeId` --> `N` everywhere, `Tx` --> `T` for `QueueingHoneyBadger`, and `Tx` --> `C` everywhere else?

**Comment by c0gent** [comment]
Closed by 697ba6f0a8c54b03236dcf05f8d0044e6f26c93c
